actionpack/lib/action_controller/metal/rack_delegation.rb

```diff
@@ -8,9 +8,15 @@ module RackDelegation
     delegate :headers, :status=, :location=, :content_type=,
              :status, :location, :content_type, :response_code, :to => "@_response"
 
-    def dispatch(action, request)
+    module ClassMethods
```

Comment:
What do you think about `new_for_env`? It reflects a constructor/initialization purpose and describes arguments without a bit of redundancy of `FooController.controller_for_env`.